src/query/ee/src/stream/handler.rs

```diff
@@ -122,34 +126,70 @@ impl StreamHandler for RealStreamHandler {
                         plan.table_database, plan.table_name
                     )));
                 }
-                options = stream.get_table_info().options().clone();
-                let stream_mode = if plan.append_only {
-                    MODE_APPEND_ONLY
+
+                let version = stream_opts
+                    .get(OPT_KEY_TABLE_VER)
+                    .ok_or_else(|| ErrorCode::IllegalStream(format!("Illegal stream '{name}'")))?
+                    .parse::<u64>()?;
+                (version, stream_opts.get(OPT_KEY_SNAPSHOT_LOCATION).cloned())
+            }
+            Some(StreamNavigation::AtPoint(point)) => {
+                let fuse_table = FuseTable::try_from_table(table.as_ref())?;
+                let source = fuse_table.navigate_to(point).await?;
+                if let Some(snapshot_loc) = source.snapshot_loc().await? {
+                    let (snapshot, _) =
+                        SnapshotsIO::read_snapshot(snapshot_loc.clone(), fuse_table.get_operator())
+                            .await?;
+                    let Some(version) = snapshot.table_version else {
+                        return Err(ErrorCode::IllegalStream(
+                            "The stream navigation at point has not table version".to_string(),
+                        ));
+                    };
+
+                    // The table version is the version of the table when the snapshot was created.
+                    // We need make sure the version greater than the table version,
+                    // and less equal than the table version after the snapshot commit.
+                    (version + 1, Some(snapshot_loc))
                 } else {
-                    MODE_STANDARD
-                };
-                options.insert(OPT_KEY_MODE.to_string(), stream_mode.to_string());
+                    unreachable!()
+                }
             }
             None => {
-                let stream_mode = if plan.append_only {
-                    MODE_APPEND_ONLY
-                } else {
-                    MODE_STANDARD
-                };
-                options.insert(OPT_KEY_MODE.to_string(), stream_mode.to_string());
-                options.insert(OPT_KEY_TABLE_NAME.to_string(), plan.table_name.clone());
-                options.insert(
-                    OPT_KEY_DATABASE_NAME.to_string(),
-                    plan.table_database.clone(),
-                );
-                options.insert(OPT_KEY_TABLE_ID.to_string(), table_id.to_string());
-                options.insert(OPT_KEY_TABLE_VER.to_string(), table_version.to_string());
                 let fuse_table = FuseTable::try_from_table(table.as_ref())?;
-                if let Some(snapshot_loc) = fuse_table.snapshot_loc().await? {
-                    options.insert(OPT_KEY_SNAPSHOT_LOCATION.to_string(), snapshot_loc);
-                }
+                (table_version, fuse_table.snapshot_loc().await?)
```

Comment:
should we use  `table_version + 1` here?  Although not incrementing by 1 should still work (hope I'm not misunderstanding), unifying this value might be better?

Reply:
Here, the "table version" refers to the current version of the table. If the table version is incremented by 1, it may point to a subsequent snapshot or version.

Reply:
Got it, thanks